server/src-lib/Hasura/RQL/Types/ComputedField.hs

```diff
@@ -29,12 +29,12 @@ fromComputedField = FieldName . computedFieldNameToText
 
 data FunctionTableArgument
   = FTAFirstArgument
-  | FTAName !FunctionArgName
+  | FTAName !FunctionArgName !Int -- ^ argument name and index
```

Comment:
Does this subsume the `FTAFirstArgument` case? It seems like `FTAFirstArgument` is now just a particular case of `FTAName` where the index is `0`.

Reply:
> Does this subsume the FTAFirstArgument case? It seems like FTAFirstArgument is now just a particular case of FTAName where the index is 0

Yes. But, if the index is not `0`, then `FunctionArgName` is a must. The first argument may or may not have a `FunctionArgName` and the argument name have no significance since it will be the first argument value and always positional. This type is directly inferred from the metadata API. If `table_argument` is not given then it is `FTAFirstArgument` else `FTAName`. And `FTAName` can be a first argument (if the index is `0`). My point here is to say if the table argument is first or it is named. :)

Reply:
Ah, I see! And the name is needed in the other cases because the other arguments might be optional? That makes sense to me. Maybe add a short comment explaining the distinction? (Also, this is a minor nit, but I think `FTAFirst` and `FTANamed` might be more consistent names for the constructors.)